examples/querying_entity_versions.ipynb

```diff
@@ -0,0 +1,782 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "0fcf0214-2561-488a-8eee-0330ca27710d",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from openassetio_mediacreation.traits.lifecycle import VersionTrait\n",
+    "from openassetio.access import ResolveAccess\n",
+    "\n",
+    "entity_data = manager.resolve(logo_ref, {VersionTrait.kId}, ResolveAccess.kRead, context)\n",
+    "\n",
+    "version_trait = VersionTrait(entity_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a6c4f0ff-0e5d-40b7-84d6-d46efa8cba7a",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "We use the `VersionTrait` class as a wrapper around the generic `TraitsData` returned from the API to inspect it's properties - which contain information about the entity's version. This helps ensure we don't accidentally mis-spell traits or their properties.\n",
+    "\n",
+    "MediaCreation use the term \"tag\" for some string that identifies the version of an entity. Using a string, rather than a number allows full flexibility in how a specific manager may wish to handle versions and dynamic \"meta-versions\".\n",
+    "\n",
+    "**Note: that the tags themselves are specific to each manager, and are not defined by the API or the MediaCreation library. Hosts should not attempt to string munge entity references, and use the methods covered in this document instead if they wish to obtain references for specific versions. Otherwise, this defeats the function of the API as an abstraction layer.**\n",
+    "\n",
+    "The `specifiedTag` property holds the version tag addressed by the entity reference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "1c0683dd-c601-41b4-a069-d5acb3c40cbf",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": [
+       "> **Result:**\n",
+       "> `latest`"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "helpers.display_result(version_trait.getSpecifiedTag())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "17049575-fc94-4ac1-9db4-466cccbe1061",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "BAL uses the `v=` query param to define versions in its references. Ours did not have one (`\"bal:///project_artwork/logos/openassetio\"`), in this case, BAL elects to return the latest version. Again note, this is a BAL specific behaviour, other managers may handle things differently.\n",
+    "\n",
+    "The version trait defines that the `specifiedTag` property should hold the tag string that could be used to achieve the same result, so is set to `\"latest\"`."
```

Comment:
```suggestion
"The version trait defines that the `specifiedTag` property should hold an unambiguous tag for a specific version of an entity. BAL uses the `v=` query param to define versions in its references. We see here that as we did not provide one (`\"bal:///project_artwork/logos/openassetio\"`), BAL has elected to default to `latest`. Again note, this is a BAL specific behaviour, other managers may handle things differently.\n",
```

[Consider] Trying to address 2 things :
 - The paragraph split here in the notebook can make it seem that the latter paragraph referencing `latest` is referring to the code block below, (rather than the correct one above,) as it's closer.
 - "the tag string that could be used to achieve the same result" I think is unhelpful and relies on people going and reading the Version trait docs to fully understand it. This way we provide the information directly.

Reply:
Hows [this](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/59/commits/523e6cda8d80dd598d148d51e864da2007eb779b) in conjunction with the other structural rejig?

Reply:
The "could be used in a subsequent query" is the key change there, before that I was reading it as "the same result as the what we spoke about in previous paragraph"